### ISSUE TITLE
Mobile: Paella player control bar too high

### ISSUE THREAD

**Issue by oas777** [elan-ev/tobira#1276]
![image](https://github.com/user-attachments/assets/4ed95f86-04ed-4769-bbc8-3db63aaf8fd0)

**Comment by oas777** [comment]
Oh, and some other Paella issues at least on my device:

- Control doesn't disappear once used.
- Pressing the "Option" symbol makes a overlay appear so that you cannot "unpress" the symbol.
- Fullscreen doesn't work.

Hope these are problems with this deployment ...

**Comment by owi92** [comment]
We could hide "show slides" and probably also make the other icons shrink when on smaller screens.
In order for control to disappear, you need to tap outside the player. If that's not a good solution, I think it would need to be fixed in paella.

I can't reproduce the Options thing on my device, so that might be specific to iOS. Fullscreen not working definitely is.

**Comment by oas777** [comment]
The control button being covered by the overly for "control" seems to be a result of the two lines created. Do we solve this
- by reducing the size of icons (not very comfortable on a small screen) or
- getting rid of icons? If so, I suggest +/- 10s could go

**Comment by LukasKalbertodt** [comment]
- Buttons in two lines: that does not happen on my iPhone 13 Mini, but I have seen this before and can easily reproduce it by shrinking the browser window size enough. Yes we should fix that. I don't mind shrinking the buttons a bit. 
  - Removing +-10s: it would be nice if Paella supported the YouTube double-tab gesture on the left/right side of the video for this functionality. Then we could easily remove those buttons.
- Control bar: can reproduce, but it doesn't bother me that much. Clicking outside of the player or on the video does fix it. But yeah would have to be fixed in Paella.
- Option button problem: also cannot reproduce, but as you said, probably result of the two lines.
- Fullscreen: Dang, how did this break again. Thanks for noticing. It's very hard to write automated tests for this. Maybe we should have a checklist of manual tests we need to perform before each release...

**Comment by oas777** [comment]
I'm about to contact UPV about the slide segmenter, is there anything else I should mention?

**Comment by LukasKalbertodt** [comment]
The problem with the two lines is even worse for videos with subtitles: https://tobira.opencast.org/v/KrlVO3Ks4bG?list=Lwiz32KAK0i

Keeping all icons in one line on our minimum supported screen width of 320px is... hard. Not sure if we can achieve this by just making everything smaller. Maybe at that point one actually has to hide some elements. 

> I'm about to contact UPV about the slide segmenter, is there anything else I should mention?

Can't think of anything right now. Maybe we need to talk to them regarding fullscreen mode, but I'm not sure yet if the bug is on our or their side. Have to investigate first.

**Comment by owi92** [comment]
> I'm about to contact UPV about the slide segmenter, is there anything else I should mention?

David asked me a while ago to open this issue: https://github.com/polimediaupv/paella-core/issues/372.
And there are two other open issues related to the slide segments (which I suspect you would mention anyway, but for completeness, here they are): https://github.com/polimediaupv/paella-slide-plugins/issues/7, https://github.com/polimediaupv/paella-slide-plugins/issues/8.

**Comment by oas777** [comment]
I'm all in favor of getting rid of icons, just like "Show slides" and maybe even "Video layout" with dual video videos. And while we're at it: Who can actually see/use those "in-video" control elements?
![image](https://github.com/user-attachments/assets/e30de78e-d42d-4b20-b780-667ee5b13d46)

**Comment by owi92** [comment]
> Who can actually see/use those "in-video" control elements?

They become opaque when you're hovering over the video, but on mobile you need to tap inside the video instead (but not the control bar, above that).

**Comment by oas777** [comment]
Let me rephrase that question: Why are we offering the full range of features Paella has in mobile, where space is limited? I know there's larger devices, but instead of making icons even smaller and "in-video" elements cluttering the UI, let's get rid of some stuff in mobile.

**Comment by owi92** [comment]
I see your point. We don't need to offer everything, but users should at least have some control over which stream they want to see.
Or we just default to showing slides only, as I suspect those will be more important than the view of the lecturer in most cases.

Please tell us which elemets you want to get rid of on mobile screens and which ones you think should be kept.

**Comment by oas777** [comment]
> We don't need to offer everything, but users should at least have some control over which stream they want to see.
They can do that video the "Video layout" button under "Options" in the control bar - so we're offering the same function twice on the size of a stamp.

**Comment by oas777** [comment]
> Please tell us which elemets you want to get rid of on mobile screens and which ones you think should be kept.

I suggested some, but would like to hear others - including yourself.

**Comment by dagraf** [comment]
Hiding buttons: I regularly use the +/- 10s buttons. Therefore, I would vote to keep them. Also because accurate scrolling in the timeline is quite hard on small screen and with clumsy fingers.

My suggestions:
- Keep and shrink a bit: Play/Pause, +/- 10s, Settings-Wheel, Fullscreen
- Move the rest to settings.

**Comment by oas777** [comment]
Adding this to July/August "required" as per (my interpretation of) our discussion today.

**Comment by oas777** [comment]
Here's what Miguel had for me, @owi92:

The issue arises because the volume bar does not fit within the available space.

To address this, the minContainerSize property should be used in plugins to ensure they do not appear below a certain size of the playback bar. Clearly, if there is no space for the volume control in the bar due to limited space, there is little that can be done.

The recommended approach would be to hide the volume plugin or other less critical plugins when the player is in a "small" state.

For instance, if it is observed that the player behaves oddly below a width of 450 pixels, the minContainerSize property for the volume plugin can be set to 450. If the volume plugin is deemed too important to hide, other plugins can be hidden instead.

Example:

"es.upv.paella.plugin": {
    "enabled": true,
    "minContainerSize": 500,
},

This prevents the plugin from displaying in the bar when the size is less than 500px

**Comment by oas777** [comment]
Thanks, Ole. @dagraf was in favor of shrinking icons (and/or moving them to settings), but I wonder what that means in terms of usability. Thouhgts?

**Comment by oas777** [comment]
>We'd only do that for smaller (i.e. mobile) screens.

Am I correct in assuming we're only doing this for mobile phones in vertical position? My iPhone fully displays the single-line control bar horizontally and I assume the iPad doesn't have a problem vertically either.

@dagraf, are you really watching videos with your phone vertically - and expecting to navigate -10s/+10s? To me, this is a bit like expecting the editor to work on an iPhone.

**Comment by dagraf** [comment]
I will not be reviewing my position (yet). This problem also occurs when a video is embedded in a multi-column page, resulting in the video being displayed in a small size.

What about discussing this issue in our next meeting with the goal of coming to a final solution?

**Comment by owi92** [comment]
Not quite. We'd have to reduce some margins as well, i.e. the space around the buttons, and/or do sth else in addition like shrink some buttons, as David suggested.

<img width="375" height="260" alt="Image" src="https://github.com/user-attachments/assets/47e68802-357d-4f43-98da-b376cfe73593" />

<img width="375" height="260" alt="Image" src="https://github.com/user-attachments/assets/69c7e055-6e2e-40c5-baef-7529ff0eba7c" />

**Comment by oas777** [comment]
Thanks. What about David's suggestion to "Keep and shrink a bit: Play/Pause, +/- 10s, Settings-Wheel, Fullscreen, Move the rest to settings", does that work?

**Comment by dagraf** [comment]
True, concerning @oas777 comment about the speed button.
